velox/connectors/hive/storage_adapters/s3fs/S3Util.h

```diff
@@ -30,13 +30,32 @@ namespace facebook::velox {
 
 namespace {
 constexpr std::string_view kSep{"/"};
+// AWS S3 EMRFS, Hadoop block storage filesystem on-top of Amazon S3 buckets.
 constexpr std::string_view kS3Scheme{"s3://"};
+// This should not be mixed with s3 nor the s3a.
+// S3A Hadoop 3.x (previous connectors "s3" and "s3n" are deprecated).
+constexpr std::string_view kS3aScheme{"s3a://"};
+// OSS Alibaba support S3 format, usage only with SSL.
+constexpr std::string_view kOssScheme{"oss://"};
 // From AWS documentation
 constexpr int kS3MaxKeySize{1024};
 } // namespace
 
+inline bool isS3AwsFile(const std::string_view filename) {
+  return filename.substr(0, kS3Scheme.size()) == kS3Scheme;
+}
+
+inline bool isS3aFile(const std::string_view filename) {
+  return filename.substr(0, kS3aScheme.size()) == kS3aScheme;
+}
+
+inline bool isOssFile(const std::string_view filename) {
+  return filename.substr(0, kOssScheme.size()) == kOssScheme;
+}
+
 inline bool isS3File(const std::string_view filename) {
-  return (filename.substr(0, kS3Scheme.size()) == kS3Scheme);
+  // TODO: Each implementation should be registered as separate filesystem
```

Comment:
nit: each comment must end with a full stop. Probably say
`// TODO: Each prefix should be implemented as its own filesystem.` ?
The registration then is implied.

Reply:
True, once again I was in a hurry. Thanks.